src/psaform/psaform.jsx

```diff
@@ -0,0 +1,212 @@
+            <Typography variant="h6">
+              {field.label} {field.required && <span style={{ color: "red" }}>*</span>}
+            </Typography>
+          </Grid>
+          <Grid item xs={12}>
+            <Typography variant="body1">{field.description}</Typography>
+          </Grid>
+          <Grid item xs={12}>
+            {field.type === "text" ? (
+              <PSATextField
+                {...field.props}
+                onChange={(event) => handleTextInputChange(event, field.name)}
+              />
+            ) : (
+              <FormGroup>
+               {field.options.map((checkbox, index) => (
+              <FormControlLabel
+                key={index}
+                control={
+                  <Checkbox
+                    {...checkbox.props}
+                    onChange={handleCheckboxChange}
+                  />
+                }
+                label={checkbox.label}
+              />
+            ))}
+              </FormGroup>
+            )}
+          </Grid>
+        </Grid>
+      ))}
+
+      <Grid container item spacing={1} justifyContent="flex-start" alignItems="flex-start">
+        {isSubmitDisabled && (
+          <Grid item xs={12}>
+            <Typography variant="body1" style={{ color: "red" }}>
+              {checkDisabled().message}. Please fill all required fields before submitting.
+            </Typography>
+          </Grid>
+        )}
+        {buttons.map((button, index) => (
+          <Grid key={index} item xs={12}>
+            <PSAButton
+              {...button.props}
+              onClick={button.action === "submit" ? handleSubmit : button.onClick}
+              disabled={isSubmitDisabled}
+            />
+          </Grid>
+        ))}
+      </Grid>
+
+      <Snackbar
+        open={snackbarData.open}
+        autoHideDuration={5000}
+        onClose={() => setSnackbarData({ ...snackbarData, open: false })}
+        message={snackbarData.message}
+        anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
+        color={snackbarData.color}
+      />
+    </Grid>
+  );
+};
+
+PSAForm.propTypes = {
+  apiUrl: PropTypes.string.isRequired,
+  submitMessage: PropTypes.string.isRequired,
+  headerTitle: PropTypes.string.isRequired,
+  fields: PropTypes.arrayOf(
+    PropTypes.shape({
+      type: PropTypes.oneOf(["text", "checkbox"]).isRequired,
+      label: PropTypes.string.isRequired,
+      description: PropTypes.string,
+      props: PropTypes.object,
+      options: PropTypes.arrayOf(
+        PropTypes.shape({
+          label: PropTypes.string,
+          props: PropTypes.object,
+        })),
+      name: PropTypes.string.isRequired,
+      required: PropTypes.bool,
+    })
+  ).isRequired,
+  buttons: PropTypes.arrayOf(
+    PropTypes.shape({
+      props: PropTypes.object,
+      action: PropTypes.string,
+      onClick: PropTypes.func,
+    })
+  ).isRequired,
+  consentRedux: PropTypes.bool,
+  pirschAnalytics: PropTypes.func.isRequired,
```

Comment:
Can you just remove the `isRequired` because not all of them should be strictly required.